my-app/src/Form.tsx

```diff
@@ -0,0 +1,150 @@
+import { Image } from './Image';
+import { Input } from './Input';
+import { Notes } from './Notes';
+import { Button } from './Button';
+import { useEffect, useState } from 'react';
+import { ImageInput } from './ImageInput';
+import { addEntry, readEntry, removeEntry, updateEntry } from './data';
+import { useNavigate, useParams } from 'react-router-dom';
+import { ModalDelete } from './ModalDelete';
+
+export function Form() {
+  const [title, setTitle] = useState<string>('');
+  const [photoUrl, setPhotoUrl] = useState(
+    'images/placeholder-image-square.jpg'
+  );
+  const [notes, setNotes] = useState('');
+  const [isOpen, setIsOpen] = useState(false);
+  const navSubmit = useNavigate();
+
+  async function handleSave() {
+    if (id === undefined) {
+      await addEntry({ title, notes, photoUrl });
+    } else if (id !== undefined) {
+      await updateEntry({ title, notes, photoUrl, entryId: Number(id) });
+    }
+    navSubmit('/');
+  }
+
+  const [isLoading, setIsLoading] = useState(true);
+  const [error, setError] = useState<unknown>();
+
+  const { id } = useParams();
+
+  useEffect(() => {
+    async function work() {
+      try {
+        const read = await readEntry(Number(id));
+        if (!read) throw new Error(`id ${id} does not exist`);
+        setTitle(read?.title);
```

Comment:
Since you throw on line 38 if `read` is not defined, you don't need the `?` in these next 3 statements.